om/CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+###############################################################################
+# Copyright (c) 2017, 2018 IBM Corp. and others
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at http://eclipse.org/legal/epl-2.0
+# or the Apache License, Version 2.0 which accompanies this distribution
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+###############################################################################
+
+omr_assert(FATAL_ERROR TEST OMR_OM)
+
+omr_assert(FATAL_ERROR
+	TEST OMR_GC # AND OMR_JITBUILDER
+	MESSAGE "Om requires the OMR_GC and OMR_JITBUILDER to be enabled"
+)
+
+omr_assert(FATAL_ERROR
+	TEST NOT OMR_EXAMPLE
+	MESSAGE "The OMR example code is temporarily incompatible with Om"
+)
+
+set(CMAKE_CXX_STANDARD 11 PARENT_SCOPE)
```

Comment:
Can this be removed?  We should not be forcing C++11 as not all compiler support this.  Extended support is allowed as long as all of the features are available on Visual Studio 10 and xlC 11 (I think it is xlC 11)

Reply:
This isn't the cleanest way to set the standard, but should be okay. This will cause cmake to choose the best flag for the compiler toolchain, to get as close as possible to c++11.  Cmake will still succeed even If the compiler doesn't support c++11. To treat this as an error, you'd turn `CMAKE_CXX_STANDARD_REQUIRED` on, but I'm not doing that here.

Lately, I have been working to restrict om to the language or stdlib features supported by our target specs (toolchain & target platform). There is more work to do here, as om has never actually been compiled for most specs (and actually used to target c++14).

On older versions of cmake, or on some toolchains, cmake is bad at picking the flag. It's not something I'm worrying about now, but will probably go wrong if we ever port Om to eg xlc.

Eventually, I would like to set the standard flag for the omr_om/omrgc target only, but that requires that the omrgc target is already defined. Due to the way I've structured the top level cmakelists.txt, that isn't yet possible. This probably needs a cleaner solution. Possibly, an OMR_GC_CXX_STANDARD variable?